### ISSUE TITLE
addr2:street / addr2:housenumber support

### ISSUE THREAD

**Issue by neonxp** [organicmaps/organicmaps#771]
Hi, guys! What is the situation with the dual addressing of corner houses in OM? As I understand it, there is no clear consensus in the OSM community on how to properly support it. But there is one option using the `addr2:street` / `addr2:housenumber` tags. It doesn't look like a great solution, but it's IMHO the best of the worst. And It would be great to see support in OM. In particular, in the search, on the object card and on the map itself in the form of a house numbers like "81/23" for case https://www.openstreetmap.org/way/71169981
fyi: https://wiki.openstreetmap.org/wiki/RU:Addresses#.D0.A3.D0.B3.D0.BB.D0.BE.D0.B2.D1.8B.D0.B5_.D0.B4.D0.BE.D0.BC.D0.B0

**Comment by biodranik** [comment]
@Zverik what do you think?

**Comment by Zverik** [comment]
`addr2:street` is not supported by any other software. `addr:street2` is also very rare to be supported, although I think maps.me / organicmaps does support it. The correct way at multiple addressing is simply to create another addressed node inside a building.

**Comment by neonxp** [comment]
Yes, I agree that tags are not supported by other software. At the moment, there is only a proposal: https://wiki.openstreetmap.org/wiki/RU:Proposed_features/AddrN.

The problem with the node inside a building is that it displays ugly (the red area in the screenshot). And, I'm sure, there are side effects associated with the fact that the polygon has its own address and it may not know about the existence of its other address on the node. Although logically both addresses are equal.

May be good way is display a double address through a slash (as in the green zone of the screenshot), if there is a node with an address in the polygon? And also have both addresses on the card?

![IMG_8177](https://user-images.githubusercontent.com/190940/124030595-02b16480-d9ff-11eb-895a-bc03018f6f11.PNG)

**Comment by biodranik** [comment]
Some addresses in Russia, for example, are using slashes, so it's a bad idea. Need to think about something better (use parentheses?)

**Comment by natrius** [comment]
Slashes are a bad idea. As far as i'm aware a lot of houses in austria do have two adresses on corners and we opted to just drop 2 adress nodes there because its easier for everyone. See this building https://www.openstreetmap.org/way/155053493 for example.

**Comment by neonxp** [comment]
Slash IMHO using often on double numbering of the corner houses. As on screenshot:
<img width="537" alt="Снимок экрана 2021-07-01 в 16 03 25" src="https://user-images.githubusercontent.com/190940/124129133-253a9080-da86-11eb-9dee-301fbc196993.png">
It seems that any solution will be good in which the house numbers will be neatly displayed next to each other and not at random points of the polygon. And if all the addresses are displayed in the house card and in the search.

**Comment by RedAuburn** [comment]
closing as not planned, it's standard practice that different addresses are mapped with either a node or internal area

**Comment by biodranik** [comment]
Closing it does not solve a proper address display when selecting a building or some feature inside this building.

In any way, our generator should detect that a building has two different addresses and properly process them, or even extract addr2: fields too, to provide better UX for end-users.
